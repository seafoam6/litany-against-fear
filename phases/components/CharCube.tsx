
interface Props {
  solution: string;
}

const CharCube: React.FC<Props> = ({ solution }) => {

  return <input type="text" minLength={1} maxLength={1} />
}

export default CharCube