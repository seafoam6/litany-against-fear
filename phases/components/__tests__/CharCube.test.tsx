import CharCube from "../CharCube";
import { render, fireEvent, waitFor, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

const setupTypedCorrect = async () => {
  const { container, debug } = render(
    <CharCube solution="s" isFocused={false} keyy={3} />
  );

  const input = container.querySelector("input");
  await userEvent.type(input, "s");
  return { input, debug, container };
};

const setupTypedWrong = async () => {
  const { container, debug } = render(
    <CharCube solution="s" isFocused={false} keyy={3} />
  );

  const input = container.querySelector("input");
  await userEvent.type(input, "x");
  return { input, debug, container };
};

const setupType = async (answer, typed) => {
  const { container, debug } = render(
    <CharCube solution={answer} isFocused={false} keyy={3} />
  );

  const input = container.querySelector("input");
  await userEvent.type(input, typed);
  return { debug, container };
};

describe("CharCube", () => {
  it("should render", () => {
    const { container } = render(
      <CharCube solution="s" isFocused={false} keyy={3} />
    );
    expect(container).toMatchSnapshot();
  });

  describe("inputing correct answer", () => {
    it("should have correct value", async () => {
      const { input } = await setupTypedCorrect();
      expect(input.value).toEqual("s");
    });
    it("should be solved", async () => {
      const { container } = await setupTypedCorrect();

      expect(container).toMatchSnapshot();
    });
    it("should be case insensitive", async () => {
      const { container } = await setupType("S", "s");
      expect(container).toMatchSnapshot();
    });
  });

  describe("inputing wrong answer", () => {
    it("should not have a value", async () => {
      const { input, debug } = await setupTypedWrong();
      expect(input.value).toEqual("");
    });
    it("should have placeholder matching typed", async () => {
      const { input, debug } = await setupTypedWrong();
      expect(input).toHaveAttribute("placeholder", "x");
    });
    it("should be not readonly", async () => {
      const { input, debug } = await setupTypedWrong();
      expect(input).not.toHaveAttribute("readonly");
    });
  });
});
