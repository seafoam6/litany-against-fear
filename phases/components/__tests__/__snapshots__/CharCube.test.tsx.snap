// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharCube inputing correct answer should be case insensitive 1`] = `
<div>
  <div
    class="jsx-2087504289"
  >
    S
  </div>
</div>
`;

exports[`CharCube inputing correct answer should be solved 1`] = `
<div>
  <div
    class="jsx-2087504289"
  >
    s
  </div>
</div>
`;

exports[`CharCube should render 1`] = `
<div>
  <input
    class="jsx-3905544307"
    maxlength="1"
    minlength="1"
    placeholder=""
    type="text"
    value=""
  />
</div>
`;
